tests/test_mpi.py

```diff
@@ -1474,60 +1501,53 @@ def test_adjoint_codegen(self, shape, kernel, space_order, nbpml, save):
         assert len(fwd_calls) == 1
         assert len(adj_calls) == 1
 
-    def run_adjoint_F(self, shape, kernel, space_order, nbpml, save,
-                      Eu, Erec, Ev, Esrca):
+    def run_adjoint_F(self, nd):
         """
         Unlike `test_adjoint_F` in test_adjoint.py, here we explicitly check the norms
         of all Operator-evaluated Functions. The numbers we check against are derived
         "manually" from sequential runs of test_adjoint::test_adjoint_F
         """
+        Eu, Erec, Ev, Esrca = self.gen_serial_norms(nd)
+        shape = self._shapes[nd]
+        so = self._so[nd]
         tn = 500.  # Final time
         nrec = 130  # Number of receivers
 
         # Create solver from preset
-        solver = acoustic_setup(shape=shape, spacing=[15. for _ in shape], kernel=kernel,
-                                nbpml=nbpml, tn=tn, space_order=space_order, nrec=nrec,
+        solver = acoustic_setup(shape=shape, spacing=[15. for _ in shape],
+                                tn=tn, space_order=so, nrec=nrec,
                                 preset='layers-isotropic', dtype=np.float64)
         # Run forward operator
-        rec, u, _ = solver.forward(save=save)
+        rec, u, _ = solver.forward()
 
-        assert np.isclose(norm(u), Eu, rtol=Eu*1.e-8)
-        assert np.isclose(norm(rec), Erec, rtol=Erec*1.e-8)
+        assert np.isclose(norm(u) / Eu, 1.0)
```

Comment:
why do you have to change tolerance?

Reply:
This is just relative error check and doesn't depend on what the value is so should be more robust.